### DISCUSSION TITLE
Docker Compose Images for GoTrue and Studio Are Outdated – How Should We Handle Updates?

### DISCUSSION THREAD

**Discussion by tsensei** [supabase/supabase#30362] [Questions]
Hi Supabase Team,

I've been working on self-hosting Supabase using the [supabase/docker/docker-compose.yml](https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml) file. I noticed that the `supabase/gotrue` image specified in the compose file is currently set to version `v2.158.1`:

```
auth:
  container_name: supabase-auth
  image: supabase/gotrue:v2.158.1
```

However, when I checked Docker Hub, there are several newer versions available, such as `v2.164.0-rc.8` (pushed just a day ago) and other stable versions as well. The same discrepancy exists for the Studio service as well.

### My concerns are:
1. **Update Frequency**: The docker-compose.yml image versions are rarely updated if I am not wrong. Is there a schedule for updating these images, or is it expected that users manually update the container versions themselves?

2. **Manual Updates Risks**: If we are supposed to update the container versions ourselves, there's uncertainty regarding:

- **New Environment Variables:** Are there any new environment variables introduced in the latest versions that we need to configure?
- **Config File Changes:** Have there been changes to configuration files or required volume mounts that need to be addressed when updating?

3. **Stability and Compatibility:** Using outdated images might lead to compatibility issues or missing out on important security updates and features.

**Questions for the Team:**

- What is the recommended approach for keeping the Docker images up-to-date in the self-hosted setup?
- Will there be regular updates to the docker-compose.yml, or should the community take on the responsibility of updating the image versions?
- Is there documentation or a changelog that outlines any necessary changes (like new environment variables or config adjustments) when updating to newer image versions?

Ensuring that the Docker Compose setup remains up-to-date is crucial for maintaining the security, stability, and feature set of a self-hosted Supabase instance. I’d appreciate guidance on the best practices for managing these updates.

Thank you for the awesome project!

**Comment by aantti** [comment]
My apologies it took so long 🙈

I've recently added:
- [CHANGELOG.md](https://github.com/supabase/supabase/blob/master/docker/CHANGELOG.md)
- [versions.md](https://github.com/supabase/supabase/blob/master/docker/versions.md) (to track image updates and for rollbacks)

Also mentioned it [here](https://github.com/orgs/supabase/discussions/39820#discussioncomment-15003158).
